src/modules/RA/DataLayer/components/DataLayerForm.js

```diff
@@ -20,75 +20,80 @@ const DataLayerForm = ({ ...props }) => {
-    },
-    errors: { fields = [], popup_config: popupConfig },
-    touched,
-  }) => {
-    const popupfieldIds = popupfields.flatMap(({ sourceFieldId }) => sourceFieldId);
-    const fieldIndexes = sourcefields
-      .map((field, index) => (
-        popupfieldIds.includes(field.sourceFieldId) ? index : null))
-      .filter(Boolean);
-    const fieldsInError = fields.filter((f, index) => f && fieldIndexes.includes(index));
-
-    let inError = false;
-    if (fieldsInError.length > 0) {
-      inError = true;
-    }
-    if (popupConfig && touched['popup_config.wizard.fields[0]']) {
-      inError = true;
-    }
-    setPopupInError(inError);
-  }, [setPopupInError]);
-
-  const onMiniSheetErrorChange = React.useCallback(({
-    values: {
-      fields: sourcefields,
-      minisheet_config: { wizard: { tree = [] } = {} } = {},
+  const onPopupErrorChange = React.useCallback(
+    ({
+      values: {
+        fields: sourcefields,
+        popup_config: { wizard: { fields: popupfields = [] } = {} } = {},
+      },
+      errors: { fields = [], popup_config: popupConfig },
+      touched,
+    }) => {
+      const popupfieldIds = popupfields.flatMap(({ sourceFieldId }) => sourceFieldId);
+      const fieldIndexes = sourcefields
+        .map((field, index) => (popupfieldIds.includes(field.sourceFieldId) ? index : null))
+        .filter(Boolean);
+      const fieldsInError = fields.filter((f, index) => f && fieldIndexes.includes(index));
+
+      let inError = false;
+      if (fieldsInError.length > 0) {
+        inError = true;
+      }
+      if (popupConfig && touched['popup_config.wizard.fields[0]']) {
+        inError = true;
+      }
+      setPopupInError(inError);
     },
-    errors: { fields = [], minisheet_config: minisheetConfig },
-    touched,
-  }) => {
-    const minisheetFieldIds = tree.flatMap(({ children, sourceFieldId }) => {
-      if (sourceFieldId) {
-        return sourceFieldId;
+    [setPopupInError],
+  );
+
+  const onMiniSheetErrorChange = React.useCallback(
+    ({
+      values: { fields: sourcefields, minisheet_config: { wizard: { tree = [] } = {} } = {} },
+      errors: { fields = [], minisheet_config: minisheetConfig },
+      touched,
+    }) => {
+      const minisheetFieldIds = tree.flatMap(({ children, sourceFieldId }) => {
+        if (sourceFieldId) {
+          return sourceFieldId;
+        }
+        const ids = children.flatMap(({ sourceFieldId: id }) => id);
+        return ids;
+      });
+      const fieldIndexes = sourcefields
+        .map((field, index) => (minisheetFieldIds.includes(field.sourceFieldId) ? index : null))
+        .filter(Boolean);
+      const fieldsInError = fields.filter((f, index) => f && fieldIndexes.includes(index));
+
+      let inError = false;
+      if (fieldsInError.length > 0) {
+        inError = true;
+      }
+      if (minisheetConfig && touched['minisheet_config.wizard.title']) {
+        inError = true;
       }
```

Comment:
This part of code is a kind of "duplicate" of the function before.
Maybe you should DRY this code with its own function

Reply:
Code not modified anymore.